rdbms/src/main/java/io/spine/server/storage/jdbc/delivery/JdbcShardedWorkRegistry.java

```diff
@@ -67,10 +70,57 @@ public JdbcShardedWorkRegistry(JdbcStorageFactory storageFactory) {
     }
 
     @Override
-    public synchronized Optional<ShardProcessingSession> pickUp(ShardIndex index, NodeId nodeId) {
-        Optional<ShardProcessingSession> picked = super.pickUp(index, nodeId);
+    public synchronized PickUpOutcome pickUp(ShardIndex index, NodeId nodeId) {
+        PickUpOutcome outcome = super.pickUp(index, nodeId);
+        if(outcome.hasAlreadyPicked()) {
+            return outcome;
+        }
+        PickUpOutcome result = doubleCheckPickUp(index, nodeId, outcome);
```

Comment:
Not sure here. Maybe there is some limitation. 
But for JDBC, why can't we use `insert` for `pickUp` instead of `upsert`. As far as I understand insert will fail if it has the same key and this can indicate `negativeOutcome`.

Reply:
Ideally we would want to use a transaction here. But it's way too complex for a vanilla RDBMS (because we don't know if it even supports transactions we need), and also, for MySQL (which transactional work is a pain-in-the-ass given Querydsl API).

Another thing is that now the API has changed. We no longer insert the record and return `Optional<...Session>`. We return an outcome telling if the shard is picked by someone else, if we were unlucky to pick it up first. See [this PR](https://github.com/SpineEventEngine/core-java/pull/1505) for more detail.